### PR TITLE
Add null dictionary source

### LINKED ISSUE #44240: Empty source for external dictionaries
**Feature**

Allow creating empty dictionaries with DDL queries.

**Use case**

This feature will allow building queries and views referencing empty dictionary with `dictGet` or 'as a table', 
even if dictionary does not exist yet.

Maybe, empty dictionaries does not make sense in 'clickhouse as a part of internal infrastructure', this will 
feature may be useful in cases when clickhouse is used as a database for small application that configures
clickhouse automatically, and some dictionaries may be optional or provided by application  user.

**Describe the solution you'd like**

Provide mock source `SOURCE(EMPTY())` for dictionaries, such that every `dictGet` call will return `null_value`.

**Describe alternatives you've considered**

Make source section optional for dictionaries.

### PR DESCRIPTION
**Pull request by mayamika**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add null source for dictionaries. Closes #44240 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Dictionaries/NullDictionarySource.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+NullDictionarySource::NullDictionarySource(const NullDictionarySource & other) : sample_block(other.sample_block)
+{
+}
+
+QueryPipeline NullDictionarySource::loadAll()
+{
+    LOG_TRACE(&Poco::Logger::get("NullDictionarySource"), "loadAll {}", toString());
+    auto source = std::make_shared<NullSource>(sample_block);
+    return QueryPipeline(source);
```
Comment (yakov-olkhovskiy) [inline src/Dictionaries/NullDictionarySource.cpp:25]:
```suggestion
    return QueryPipeline(std::make_shared<NullSource>(sample_block));
```

**Review by yakov-olkhovskiy — APPROVED** [review]
would be nice to also have some tests for this

File: src/Dictionaries/NullDictionarySource.cpp (hunk 1 above)
Comment (mayamika) [reply src/Dictionaries/NullDictionarySource.cpp]: Done

**Comment by mayamika** [comment]
> would be nice to also have some tests for this

Done(???)

**Comment by yakov-olkhovskiy** [comment]
> Done(???)

yes, just let's wait CI done it job and then I'll merge

**Pull request merged**